src/lib_midi_random.c

```diff
@@ -0,0 +1,19 @@
+#include "global.h"
```

Comment:
idk if this file is necessary, can't it be combined into the file adjacent to it even if it doesn't "fully" fit

Reply:
Mangling the file structure to get static variables in the correct addresses has been a major pain for this project. Beyond that, I do believe it's normal to have source files with one very specific purpose in C, especially within libraries. It'll hopefully seem less odd once the source files are put into their own subfolder.

Reply:
For some reason, arrays are sometimes (and only sometimes) given padding after themselves, and the inconsistency of this is for some reason related to all previous static variables within the same object file. It becomes really obnoxious when I need an array on a specific address, but the compiler/linker insists on it being placed 4 bytes before that, and adding an extra static variable for padding will push the array forward by _8_ bytes for seemingly no reason. This is why the PCM buffer (and related static variables) currently still rely on the linker. Just to note for the future when either one of us attempts to tackle that issue.

Reply:
OHHH i think i remember running into this issue and figuring out what the padding rules are. basically the rule is, if its an array, it is 8 byte aligned within the file itself. so if the first static variable in the file's address ends with 0 or 8, all arrays will be padded to end with 0 or 8, but if the first static variable ends in 4 or C all arrays will be padded to end in 4 or C. this can be used to determine file splits more accurately, i remember it giving me a ton of grief

Reply:
Ohh, I see. I might look into this a little more with that in mind. That's very helpful, thank you!